test-engineering/load-tests/load_test_setup.sh

```diff
@@ -0,0 +1,101 @@
+        rm -rf $DIRECTORY/google-cloud-sdk-364.0.0-darwin-x86_64.tar.gz
+        echo -e "==================== Install SDK "
+        ./$SDK_DIRECTORY/install.sh
+        ;;
+    [nN] | [n|N][O|o] )
+        echo -e "==================== Skip GCP SDK Installation "
+        ;;
+    *) echo -e "==================== Invalid input it should be [yes or no] "
+        ;;
+esac
+
+#Setup environment
+GCLOUD=$(which gcloud)
+SED=$(which sed)
+KUBECTL=$(which kubectl)
+
+#RED_OUPUT='\033[0;31m'
+DIRECTORY=$(pwd)
+SDK_DIRECTORY=google-cloud-sdk
+MERINO_DIRECTORY=$DIRECTORY/kubernetes-config
+MASTER_FILE=locust-master-controller.yml
+WORKER_FILE=locust-worker-controller.yml
+SERVICE_FILE=locust-master-service.yml
+
+#Declare variables
+LOCUST_IMAGE_TAG=$(git log -1 --pretty=format:%h)
+#LOCUST_IMAGE_TAG=qwe123
+echo "Docker image tag for locust is set to: ${LOCUST_IMAGE_TAG}"
+
+##Kubernetes Manifests variables
+KINTO_SERVER_URL=firefox.settings.services.mozilla.com
+KINTO_COLLECTION=quicksuggest
+LOCUST_MASTER_NODE_HOST=locust-master
+KINTO_BUCKET=main
+CLUSTER=merino-load-test
+TARGET=stage.merino.nonprod.cloudops.mozgcp.net
+SCOPE="https://www.googleapis.com/auth/cloud-platform"
+REGION=us-central1
+ZONE=${REGION}-b
+PROJECT=spheric-keel-331521
+
+#Configure Kubernetes
+echo -e "==================== Prepare environments with set of environment variables "
+echo -e "==================== Set Kubernetes Cluster "
+export CLUSTER=$CLUSTER
+echo -e "==================== Set Kubernetes TARGET "
+export TARGET=$TARGET
+echo -e "==================== Set SCOPE "
+export SCOPE=$SCOPE
+
+echo -e "==================== Prepare GCP by updating default zone and project ID "
+$GCLOUD config set compute/zone $ZONE
+$GCLOUD config set project $PROJECT
+
+echo -e "==================== Refresh Kubeconfig at path ~/.kube/config "
+$GCLOUD container clusters get-credentials $CLUSTER --zone $ZONE --project $PROJECT
+
+##Build Docker Images
+echo -e "==================== Build the Docker image and store it in your project's container registry. Tag with the latest commit hash "
+$GCLOUD builds submit --tag gcr.io/$PROJECT/locust-tasks:$LOCUST_IMAGE_TAG
+echo -e "==================== Verify that the Docker image is in your project's container repository"
+$GCLOUD container images list | grep locust-tasks
+
+##Deploying the Locust master and worker nodes
+echo -e "==================== Update Kubernetes Manifests "
+echo -e "==================== Replace the target host and project ID with the deployed endpoint and project ID in the locust-master-controller.yaml and locust-worker-controller.yaml files"
+$SED -i -e "s/\[TARGET_HOST\]/$TARGET/g" $MERINO_DIRECTORY/$MASTER_FILE
+$SED -i -e "s/\[TARGET_HOST\]/$TARGET/g" $MERINO_DIRECTORY/$WORKER_FILE
+$SED -i -e "s/\[PROJECT_ID\]/$PROJECT/g" $MERINO_DIRECTORY/$MASTER_FILE
+$SED -i -e "s/\[PROJECT_ID\]/$PROJECT/g" $MERINO_DIRECTORY/$WORKER_FILE
+$SED -i -e "s/\[LOCUST_IMAGE_TAG\]/$LOCUST_IMAGE_TAG/g" $MERINO_DIRECTORY/$MASTER_FILE
```

Comment:
And I think we're missing some `sed`s. I did a quick eyeball of the .yml files above and found the following placeholders:

```
test-engineering/load-tests/kubernetes-config/locust-master-controller.yml
[KINTO_BUCKET]
[KINTO_COLLECTION]
[KINTO_SERVER_URL]
[LOCUST_IMAGE_TAG]
[PROJECT_ID]
[TARGET_HOST]


test-engineering/load-tests/kubernetes-config/locust-worker-controller.yml
[KINTO_BUCKET]
[KINTO_COLLECTION]
[KINTO_SERVER_URL]
[LOCUST_IMAGE_TAG]
[LOCUST_MASTER_NODE_HOST]
[PROJECT_ID]
[TARGET_HOST]
```

So it looks like 12-13 separate `sed` commands (gross), although I'll argue that we just hardcode the `LOCUST_MASTER_NODE_HOST=locust-master` value into the locust-worker-controller.yml since that's hardcoded everywhere in the .yml files already and it saves us a `sed` and one env var.

I wonder if we could just do the substitutions using Python or something else. Although it might mean we need to `export` all the variables as env vars and then just use Python's `os.getenv` or something.

Reply:
```suggestion
$SED -i -e "s/\[LOCUST_IMAGE_TAG\]/$LOCUST_IMAGE_TAG/g" $MERINO_DIRECTORY/$MASTER_FILE
$SED -i -e "s/\[LOCUST_IMAGE_TAG\]/$LOCUST_IMAGE_TAG/g" $MERINO_DIRECTORY/$WORKER_FILE
$SED -i -e "s/\[KINTO_BUCKET\]/$KINTO_BUCKET/g" $MERINO_DIRECTORY/$MASTER_FILE
$SED -i -e "s/\[KINTO_BUCKET\]/$KINTO_BUCKET/g" $MERINO_DIRECTORY/$WORKER_FILE
$SED -i -e "s/\[KINTO_COLLECTION\]/$KINTO_COLLECTION/g" $MERINO_DIRECTORY/$MASTER_FILE
$SED -i -e "s/\[KINTO_COLLECTION\]/$KINTO_COLLECTION/g" $MERINO_DIRECTORY/$WORKER_FILE
$SED -i -e "s/\[KINTO_SERVER_URL\]/$KINTO_SERVER_URL/g" $MERINO_DIRECTORY/$MASTER_FILE
$SED -i -e "s/\[KINTO_SERVER_URL\]/$KINTO_SERVER_URL/g" $MERINO_DIRECTORY/$WORKER_FILE
```

Need to set `LOCUST_IMAGE_TAG` in the `$WORKER_FILE` as well, plus all the `KINTO_*` variables.